app/components/contents_component.rb

```diff
@@ -21,4 +21,8 @@ def upload_csv?
   def download_csv?
     !user_version_view?
   end
+
+  def structural_link_path
+    user_version_view? ? item_user_version_structure_path(@document.id, user_version) : item_structure_path(@view_token)
```

Comment:
I think @view_token is needed for item_user_version_structure_path, probably the second parameter but including both parameters.